src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -177,4 +188,42 @@ private function registerSenders(ContainerBuilder $container)
 
         $container->getDefinition('messenger.sender_locator')->replaceArgument(0, $senderLocatorMapping);
     }
+
+    private function registerBusToCollector(ContainerBuilder $container, string $busId, array $tag)
+    {
+        if (!$container->hasDefinition($debugBusId = 'debug.traced.'.$busId)) {
+            return;
+        }
+
+        if ($container->getParameter('kernel.debug')) {
+            $container->getDefinition('messenger.data_collector')->addMethodCall('registerBus', array($tag['name'] ?? $debugBusId, new Reference($debugBusId)));
+        } else {
+            $container->removeDefinition($debugBusId);
+        }
+    }
+
+    private function registerBusMiddlewares(ContainerBuilder $container, string $busId, array $middlewares)
+    {
+        $container->getDefinition($busId)->replaceArgument(0, array_map(function (string $name) use ($container, $busId) {
+            if (!$container->has($messengerMiddlewareId = 'messenger.middleware.'.$name)) {
+                $messengerMiddlewareId = $name;
+            }
+
+            if (!$container->has($messengerMiddlewareId)) {
+                throw new RuntimeException(sprintf('Invalid middleware "%s": define such service to be able to use it.', $name));
+            }
+
+            if ('messenger.middleware.validation' === $messengerMiddlewareId && !$container->has('validator')) {
+                throw new RuntimeException('The Validation middleware is only available when the Validator component is installed and enabled. Try running "composer require symfony/validator".');
```

Comment:
would be nice if we can improve such error messages at the DI service level

Reply:
I think it's a wider question within core. That's the best way I found to keep the great DX while using this middleware without the Validator component. I would keep it here.

Reply:
See #27004

Reply:
IMHO, this still belongs to the Fwb extension where we have all the keys to detect if validation is installed and enabled. The `messenger.middleware.validation` is really specific to the Fwb wiring, so the pass alone doesn't have to deal with this unless strictly necessary.

Reply:
I see the point that this has little to do in here. Moved to the FrameworkBundle 👍